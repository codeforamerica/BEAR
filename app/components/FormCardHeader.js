import React, { Component } from 'react';

type Props = {
  children: string,
  helpText?: string
};

export default class FormCardHeader extends Component<Props, State> {
  static defaultProps = {
    helpText: null
  };

  render() {
    const { children, helpText } = this.props;
    let helpTextDiv;

    if (helpText) {
      helpTextDiv = <div className="text--help">{helpText}</div>;
    }

    return (
      <header className="form-card__header">
        <h1 className="form-card__title">{children}</h1>
        {helpTextDiv}
      </header>
    );
  }
}
