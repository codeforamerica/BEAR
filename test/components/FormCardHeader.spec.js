import sinon from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import FormCardHeader from '../../app/components/FormCardHeader';

Enzyme.configure({ adapter: new Adapter() });
const sandbox = sinon.createSandbox();

function setup(helpText) {
  const component = shallow(<FormCardHeader helpText={helpText} />);
  return { component };
}

afterEach(() => {
  sandbox.restore();
});

describe('FormCardHeader component', () => {
  it('should match exact snapshot', () => {
    const component = (
      <div>
        <FormCardHeader>Hello! This is a title.</FormCardHeader>
      </div>
    );

    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('with optional helper text', () => {
    it('displays the help text', () => {
      const { component } = setup('my help text');
      const helpDiv = component.find('.text--help');

      expect(helpDiv.text()).toEqual('my help text');
    });

    it('should match exact snapshot', () => {
      const component = (
        <div>
          <FormCardHeader helpText="this is some helper text">
            Hello! This is a title.
          </FormCardHeader>
        </div>
      );

      const tree = renderer.create(component).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
