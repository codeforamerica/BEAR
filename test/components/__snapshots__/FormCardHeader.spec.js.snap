// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCardHeader component should match exact snapshot 1`] = `
<div>
  <header
    className="form-card__header"
  >
    <h1
      className="form-card__title"
    >
      Hello! This is a title.
    </h1>
  </header>
</div>
`;

exports[`FormCardHeader component with optional helper text should match exact snapshot 1`] = `
<div>
  <header
    className="form-card__header"
  >
    <h1
      className="form-card__title"
    >
      Hello! This is a title.
    </h1>
    <div
      className="text--help"
    >
      this is some helper text
    </div>
  </header>
</div>
`;
